packages/expo-device/ios/UIDevice.swift

```diff
@@ -0,0 +1,262 @@
+        return ExpoDeviceType(modelName: "iPad Pro (9.7-inch)", deviceYearClass: 2016)
+      case "iPad7,3", "iPad7,4":
+        return ExpoDeviceType(modelName: "iPad Pro (10.5-inch)", deviceYearClass: 2017)
+      case "iPad8,1", "iPad8,2", "iPad8,3", "iPad8,4":
+        return ExpoDeviceType(modelName: "iPad Pro (11-inch) (1st generation)", deviceYearClass: 2018)
+      case "iPad8,9", "iPad8,10":
+        return ExpoDeviceType(modelName: "iPad Pro (11-inch) (2nd generation)", deviceYearClass: 2020)
+      case "iPad13,4", "iPad13,5", "iPad13,6", "iPad13,7":
+        return ExpoDeviceType(modelName: "iPad Pro (11-inch) (3rd generation)", deviceYearClass: 2021)
+      case "iPad6,7", "iPad6,8":
+        return ExpoDeviceType(modelName: "iPad Pro (12.9-inch) (1st generation)", deviceYearClass: 2015)
+      case "iPad7,1", "iPad7,2":
+        return ExpoDeviceType(modelName: "iPad Pro (12.9-inch) (2nd generation)", deviceYearClass: 2017)
+      case "iPad8,5", "iPad8,6", "iPad8,7", "iPad8,8":
+        return ExpoDeviceType(modelName: "iPad Pro (12.9-inch) (3rd generation)", deviceYearClass: 2018)
+      case "iPad8,11", "iPad8,12":
+        return ExpoDeviceType(modelName: "iPad Pro (12.9-inch) (4th generation)", deviceYearClass: 2020)
+      case "iPad13,8", "iPad13,9", "iPad13,10", "iPad13,11":
+        return ExpoDeviceType(modelName: "iPad Pro (12.9-inch) (5th generation)", deviceYearClass: 2021)
+      case "AppleTV5,3":
+        return ExpoDeviceType(modelName: "Apple TV HD (4th Generation, Siri)", deviceYearClass: 2015)
+      case "AppleTV6,2":
+        return ExpoDeviceType(modelName: "Apple TV 4K", deviceYearClass: 2017)
+      case "i386", "x86_64", "arm64":
+        return ExpoDeviceType(modelName: "Simulator iOS", deviceYearClass: nil)
+      default:
+        return ExpoDeviceType(modelName: identifier, deviceYearClass: nil)
+      }
+      #elseif os(tvOS)
+      switch identifier {
+      case "AppleTV5,3":
+        return ExpoDeviceType(modelName: "Apple TV HD (4th Generation, Siri)", deviceYearClass: 2015)
+      case "AppleTV6,2":
+        return ExpoDeviceType(modelName: "Apple TV 4K", deviceYearClass: 2017)
+      case "i386", "x86_64":
+        return ExpoDeviceType(modelName: "Simulator tvOS", deviceYearClass: nil)
+      default:
+        return identifier
+      }
+      #endif
+    }
+
+    return mapToDevice(identifier: modelIdentifier)
+  }()
+
+  // Credit: https://github.com/developerinsider/isJailBroken/blob/master/IsJailBroken/Extension/UIDevice%2BJailBroken.swift
+  var isSimulator: Bool {
+    return TARGET_OS_SIMULATOR != 0
+  }
+
+  var isJailBroken: Bool {
+    if UIDevice.current.isSimulator {
+      return false
+    }
+    if JailbreakHelper.hasCydiaInstalled() {
+      return true
+    }
+    if JailbreakHelper.doesContainSuspiciousApps() {
+      return true
+    }
+    if JailbreakHelper.doesSuspiciousSystemPathExist() {
+      return true
+    }
+    return JailbreakHelper.canEditSystemFiles()
```

Comment:
I guess it would look better if we combine all these `if X { return true }` into just one condition 😄

Reply:
Agreed, I've made a similar change :)